File: AnswerMe/sdk/AnswerMeObjC/AnswerMeObjC/AnswerMeViewController.m
/*
 * Copyright (C) 2015 RoboVM AB
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#import <AnswerMeSDK/AnswerMeSDK.h>
#import "AnswerMeViewController.h"

@interface AnswerMeViewController ()


@end

@implementation AnswerMeViewController {
  AMAnswerMeSDK* sdk;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self->sdk = [AMAnswerMeSDK instance];
    
    Class counterClass = NSClassFromString(@"AMCounterImpl");
    
    self.counter = [[counterClass alloc] init];
    
    self.currentCountLabel.text = @"Current Count";
    self.currentCount.text = @"0";
}

- (void) updateLabel {
    int value = [self.counter value];
    self.currentCount.text = [NSString stringWithFormat:@"%d", value];
}

- (IBAction)addPressed:(id)sender {
    [self.counter addValue:1];
    [self updateLabel];
}
- (IBAction)minusPressed:(id)sender{
    [self.counter minusValue:1];
    [self updateLabel];
}

@end

File: AnswerMe/sdk/AnswerMeSwift/AnswerMeSwift/AnswerMeViewController.swift
//
//  ViewController.swift
//  AnswerMeSwift
//
//  Created by Tom Wojciechowski on 28/09/2017.
//  Copyright © 2017 mobivm. All rights reserved.
//

import UIKit
import ObjectiveC

@objc
class AMCounterImpl : NSObject {
    
}

class AnswerMeViewController: UIViewController {
    
    var sdk: AMAnswerMeSDK!;
    var counter: AMCounter!;
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        sdk = AMAnswerMeSDK.instance();
        
        counter = sdk.obtainAMCounter();
        
        currentCountLabel.text = "Current Count";
        currentCount.text = "0";
        
        
        //Issues, canot cast from the implemtation to the 'abstract class'
        //let counterClazz = NSClassFromString("AMCounterImpl") as! AMCounter.Type;
        //let counterObject = counterClazz.init();
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
   func updateLabel() {
        let value = counter.value;
        currentCount.text = String(value);
    }


    @IBOutlet weak var currentCount: UILabel!
    @IBOutlet weak var currentCountLabel: UILabel!
    
    @IBOutlet weak var minusButton: UIButton!
    @IBAction func minusPressed(_ sender: Any) {
        counter.minusValue(1);
        updateLabel();
    }
    @IBOutlet weak var addButton: UIButton!
    @IBAction func addPressed(_ sender: Any) {
        counter.addValue(1);
        updateLabel();
    }
}

